Mathlib/CategoryTheory/Category/RelCat.lean

```diff
@@ -35,4 +39,128 @@ instance rel : LargeCategory RelCat where
   comp f g x z := ∃ y, f x y ∧ g y z
 #align category_theory.rel CategoryTheory.rel
 
+@[ext] theorem rel_ext {X Y : RelCat} (f g : X ⟶ Y) (h : ∀ a b, f a b ↔ g a b) : f = g :=
+  funext₂ (fun a b => propext (h a b))
+
+namespace RelCat.Hom
+
+protected theorem rel_id (X : RelCat) : 𝟙 X = (· = ·) := rfl
+
+protected theorem rel_comp {X Y Z : RelCat} (f : X ⟶ Y) (g : Y ⟶ Z) : f ≫ g = Rel.comp f g := rfl
+
+@[simp] theorem rel_id_apply₂ (X : RelCat) (x y : X) : (𝟙 X) x y ↔ x = y := by
+  rw [RelCat.Hom.rel_id]
+
+@[simp] theorem rel_comp_apply₂ {X Y Z : RelCat} (f : X ⟶ Y) (g : Y ⟶ Z) (x : X) (z : Z) :
+    (f ≫ g) x z ↔ ∃ y, f x y ∧ g y z := by rfl
+
+end RelCat.Hom
+
+/-- The essentially surjective faithful embedding
+from the category of types and functions into the category of types and relations. -/
+def graphFunctor : Type u ⥤ RelCat.{u} where
+  obj X := X
+  map f := f.graph
+
+@[simp] theorem graphFunctor_map {X Y : Type u} (f : X ⟶ Y) (x : X) (y : Y) :
+    graphFunctor.map f x y ↔ f x = y := f.graph_def x y
+
+instance graphFunctor_faithful : Faithful graphFunctor where
+  map_injective h := Function.graph_injective h
+
+instance graphFunctor_essSurj : EssSurj graphFunctor :=
+    graphFunctor.essSurj_of_surj Function.surjective_id
+
+open Classical
```

Comment:
Here, it may not change anything, but it is usually considered better to insert `classical` only in the proof.